app/services/manage_action.rb

```diff
@@ -1,91 +1,85 @@
 # frozen_string_literal: true
 
 class ManageAction
-  attr_reader :params
-
   def self.create(params, extra_params: {}, skip_queue: false, skip_counter: false)
     new(params, extra_params: extra_params, skip_queue: skip_queue, skip_counter: skip_counter).create
   end
 
   def initialize(params, extra_params: {}, skip_queue: false, skip_counter: false)
-    @params = params
+    @params = params.clone
     @skip_queue = skip_queue
     @skip_counter = skip_counter
-    @extra_attrs = extra_params
+    @extra_attrs = extra_params.clone
+    sanitize_params
   end
 
   def create
-    if multiple_actions_allowed?
-      build_action(@extra_attrs)
+    if should_create_new_action?
+      create_action
     else
+      # TODO: Update existing member
```

Comment:
There's still a todo here. If you need to update the existing member in any case independent of whether a new action is created and since it just needs the `existing_member` and `@params`, can't you call `update_existing_member` before going into this conditional block and move it out of `create_action`?

Reply:
That's a good idea, but before doing that I wanted to check with the team if this is desired behaviour. Up until now we didn't update the user if he had already taken action in the page, so doing this would change that behaviour.